paderbox/io/download.py

```
@@ -1,184 +1,142 @@
+    assert local_file.exists(), local_file
+    if target_dir is None:
+        target_dir = local_file.parent
+    else:
+        target_dir = Path(target_dir)
+    if not target_dir.exists():
+        target_dir.mkdir(parents=True, exist_ok=True)
+    if local_file.name.endswith('.zip'):
+        with zipfile.ZipFile(local_file, "r") as z:
+            # Start extraction
+            members = z.infolist()
+            for i, member in enumerate(members):
+                target_file = target_dir / member.filename
+                if not target_file.exists():
+                    try:
+                        z.extract(member=member, path=target_dir)
+                    except KeyboardInterrupt:
+                        # Delete latest file, since most likely it
+                        # was not extracted fully
+                        if target_file.exists():
+                            os.remove(target_file)
+                        raise
+                elif not exist_ok:
+                    raise FileExistsError(target_file)
+        os.remove(local_file)
+
+    elif local_file.name.endswith('.tar.gz') or local_file.name.endswith('.tar'):
+        mode = "r:gz" if local_file.name.endswith('.tar.gz') else "r"
+        with tarfile.open(local_file, mode) as tar:
+            for i, tar_info in enumerate(tar):
+                target_file = target_dir / tar_info.name
+                if not target_file.exists():
+                    try:
+                        tar.extract(tar_info, target_dir)
+                    except KeyboardInterrupt:
+                        # Delete latest file, since most likely it
+                        # was not extracted fully
+                        if target_file.exists():
+                            os.remove(target_file)
+                        raise
+                elif not exist_ok:
+                    raise FileExistsError(target_file)
+                tar.members = []
+        os.remove(local_file)
+
```

Is there a reason, that you didn't define an else path with a raise? If not, could you raise a warning, when there is an unsupported suffix?